destination.go

```diff
@@ -166,6 +166,7 @@ func (a *destinationPluginAdapter) Open(ctx context.Context, _ pconnector.Destin
 
 func (a *destinationPluginAdapter) Run(ctx context.Context, stream pconnector.DestinationRunStream) error {
 	ctx = internal.Enrich(ctx, a.cfg)
+	a.writeStrategy.SetStream(stream.Server())
 
 	for stream := stream.Server(); ; {
 		batch, err := stream.Recv()
```

Comment:
Not quite. Conduit will send a batch to the connector, and depending on the strategy, the destination will either write the batch _as is_ (single write batch) or collect even more batches (batch write) before flushing it to the connector's `Write` function.

This is done for legacy reasons. Conduit currently sends records to the destination one by one (batches of size 1), so if you want to batch your writes, you need to configure batching on the destination. However, with the pipeline architecture v2, Conduit can send a batch of multiple records to the connector, so additional batching on the destination doesn't even make sense anymore, it's much better to build the batch in the source connector.

Since we want the connector SDK to be backwards compatible, I didn't remove the destination middleware just yet, but the end goal is to completely remove it and simply flush the batch that's received from Conduit. We can do that once the v2 architecture is the default.